internal/provider/service_resource.go

```diff
@@ -623,6 +725,57 @@ func mapResponseToServiceModel(ctx *context.Context, service *statuspal.Service,
 		)
 	}
 
+	// Prepare the monitoring options if applicable
+	var monitoringOptionsData types.Object
+
+	// Define the schema for monitoring options
+	monitoringOptionsSchema := map[string]attr.Type{
+		"method":       types.StringType,
+		"keyword_up":   types.StringType,
+		"keyword_down": types.StringType,
+		"headers": types.ListType{
+			ElemType: types.ObjectType{
+				AttrTypes: map[string]attr.Type{"key": types.StringType, "value": types.StringType},
+			},
+		},
+	}
```

Comment:
Here we can do:
```suggestion
	// Define the schema for monitoring options
	monitoringOptionsSchema := map[string]attr.Type{
		"method":       types.StringType,
		"keyword_up":   types.StringType,
		"keyword_down": types.StringType,
		"headers": types.ListType{
			ElemType: types.ObjectType{
				AttrTypes: map[string]attr.Type{"key": types.StringType, "value": types.StringType},
			},
		},
	}

	// Prepare the monitoring options if applicable
	monitoringOptionsData := types.ObjectNull(monitoringOptionsSchema)
```
and later we don't need the `else` statement:
```go
	} else {
		// If MonitoringOptions is nil, create a null object
		monitoringOptionsData = types.ObjectNull(monitoringOptionsSchema)
	}
```